backend/src/main/java/com/google/sps/datastore/ParticipantDatastore.java

```diff
@@ -0,0 +1,150 @@
+
+  // Datastore Key/Property constants
+  private static final String KIND_PARTICIPANT = "Participant";
+  private static final String PROPERTY_USERNAME = "username";
+  private static final String PROPERTY_STARTTIMEAVAILABLE = "startTimeAvailable";
+  private static final String PROPERTY_ENDTIMEAVAILABLE = "endTimeAvailable";
+  private static final String PROPERTY_DURATION = "duration";
+  private static final String PROPERTY_CURRENTMATCHID = "currentMatchId";
+  private static final String PROPERTY_TIMESTAMP = "timestamp";
+
+  /** Formatter for converting between String and ZonedDateTime */
+  private static final DateTimeFormatter formatter = DateTimeFormatter.ISO_ZONED_DATE_TIME;
+
+  /** Datastore */
+  private final DatastoreService datastore;
+
+  /** Constructor */
+  public ParticipantDatastore(DatastoreService datastore) {
+    this.datastore = datastore;
+  }
+
+  /** Add Participant to datastore */
+  public void addParticipant(Participant participant) {
+    // Set properties of entity
+    Entity participantEntity = new Entity(KIND_PARTICIPANT, participant.getUsername());
+    participantEntity.setProperty(PROPERTY_USERNAME, participant.getUsername());
+    participantEntity.setProperty(
+        PROPERTY_STARTTIMEAVAILABLE, participant.getStartTimeAvailable().format(formatter));
+    participantEntity.setProperty(
+        PROPERTY_ENDTIMEAVAILABLE, participant.getEndTimeAvailable().format(formatter));
+    participantEntity.setProperty(PROPERTY_DURATION, participant.getDuration());
+    participantEntity.setProperty(PROPERTY_CURRENTMATCHID, participant.getCurrentMatchId());
+    participantEntity.setProperty(PROPERTY_TIMESTAMP, participant.getTimestamp());
+
+    // Insert entity into datastore
+    datastore.put(participantEntity);
+  }
+
+  /** Return Participant Entity from username */
+  private Entity getEntityFromUsername(String username) {
+    Key participantKey = KeyFactory.createKey(KIND_PARTICIPANT, username);
+    try {
+      return datastore.get(participantKey);
+    } catch (EntityNotFoundException e) {
+      // TODO: what error to send
+      return null;
+    }
+  }
+
+  /** Update Participant with new matchId, null out availability to show currently matched */
+  public void updateNewMatch(String username, long matchId) {
+    Entity participantEntity = getEntityFromUsername(username);
+
+    participantEntity.setProperty(PROPERTY_CURRENTMATCHID, matchId);
+
+    // Null out availibility fields if not already
+    participantEntity.setProperty(PROPERTY_STARTTIMEAVAILABLE, null);
+    participantEntity.setProperty(PROPERTY_ENDTIMEAVAILABLE, null);
+    participantEntity.setProperty(PROPERTY_DURATION, 0);
+
+    // Overwrite existing entity in datastore
+    datastore.put(participantEntity);
+  }
+
+  /** Return Participant from username */
+  @Nullable
+  public Participant getParticipantFromUsername(String username) {
+    return getParticipantFromEntity(getEntityFromUsername(username));
```

Comment:
Similar comment here about potentially adding a null check. In the case where this does return null, will the algorithm still behave expectedly? Maybe adding a test for this case might be helpful, what do you think?